### ISSUE TITLE
Calling dataframe column produces an ndarray not a series as in stock pandas

### ISSUE THREAD

**Issue by aunell1** [modin-project/modin#3302]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **Modin version** (`modin.__version__`): 0.10.1
- **Python version**:  3.7.11 Intel distribution
- **Code we can use to reproduce**:
Please contact alyssa.unell@intel.com for more information regarding code and dataset.
sub.retweet = (1-wt2[0]) * sub.retweet.values + wt2[0] * sub0.iloc[:,2].values

### Describe the problem
When we have a dataframe with column name of retweet, calling dataframe.retweet in stock pandas produces a series object while this same call in modin.pandas produces an np.ndarray object. These objects have different methods, and as such, the implementation breaks.

### Source code / logs

Traceback (most recent call last):
  File "deeplearning/RecSys2020/02_ModelsCompetition/XX_Ensemble/Ensemble.py", line 77, in <module>
    sub.retweet = (1-wt2[0]) * sub.retweet.values + wt2[0] * sub0.iloc[:,2].values
AttributeError: 'numpy.ndarray' object has no attribute 'values'

**Comment by YarShev** [comment]
Hi @aunell1 , thanks for the report! I guess the error either occurs earlier or when calling `sub0.iloc[:,2].values` because the following snippet works as it should.

```python
import modin.pandas as pd
df = pd.DataFrame({"a": [1,2], "b": [3,4]})
df.a
0    1
1    2
Name: a, dtype: int64
df.a.values
array([1, 2], dtype=int64)
df.b
0    3
1    4
Name: b, dtype: int64
df.b.values
array([3, 4], dtype=int64)
```

Can you clarify the place of the error, please?

**Comment by aunell1** [comment]
It would appear that the operations produce an np.ndarray, but in stock pandas, when this new array gets assigned to sub.retweet, future references to sub.retweet still are of the type series but in modin.pandas, future references are of the type np.ndarray.

**Comment by YarShev** [comment]
@aunell1 , thanks for clarification! I can see the issue on master. We will look into it.

```python
import modin.pandas as pd
df = pd.DataFrame({"a": [1,2], "b": [3,4]})
df.a
0    1
1    2
Name: a, dtype: int64
df.a = np.arange(2)
df.a
array([0, 1])
```

**Comment by YarShev** [comment]
@aunell1 , as a workaround for now you can replace appealing to a DF column via `.retweet` to `["retweet"]`, That should work.

```python
import modin.pandas as pd
import numpy as np
df = pd.DataFrame({"a": [1,2], "b": [3,4]})
df["a"] = np.arange(2)
df["a"]
0    0
1    1
Name: a, dtype: int32
```

**Comment by RehanSD** [comment]
Resolved by #4621
